lib/spack/spack/test/concretize_preferences.py

```diff
@@ -116,9 +116,23 @@ def test_preferred_providers(self):
 
     def test_develop(self):
         """Test concretization with develop version"""
-        spec = Spec('builtin.mock.develop-test')
+        spec = Spec('develop-test')
         spec.concretize()
         assert spec.version == spack.spec.Version('0.2.15')
+        spec = Spec('develop-test2')
+        spec.concretize()
+        assert spec.version == spack.spec.Version('0.2.15')
+
+        # now add packages.yaml with develop version
+        conf = syaml.load("""\
+develop-test:
```

Comment:
`print(conf)` looks reasonable: 
```
{'develop-test': {'version': ['develop'], 'paths': {}, 'providers': {}, 'modules': {}, 'buildable': True, 'compiler': []}, 'develop-test2': {'version': ['0.2.15.develop'], 'paths': {}, 'providers': {}, 'modules': {}, 'buildable': True, 'compiler': []}}
```
(this is for the current version of the PR with two packages in `packages.yaml`)